src/NumericalAlgorithms/Spectral/Spectral.hpp

```diff
@@ -72,23 +72,31 @@ std::ostream& operator<<(std::ostream& os,
                          const Quadrature& quadrature) noexcept;
 /// \endcond
 
+namespace detail {
+constexpr size_t minimum_number_of_points(
+    const Basis basis, const Quadrature quadrature) noexcept {
+  // Silence compiler warnings.
+  (void)basis;
+  (void)quadrature;
+  if (quadrature == Quadrature::Gauss) {
```

Comment:
Another clarification question: The previous version appeared to produce a compile-time value for the appropriate basis. If my understanding of `if`s with `constexpr` is correct, this no longer produces a compile-time value if the basis is valid; is that right?
However, it seems it didn't generate any errors in Travis, so either I'm wrong or we never used them at compile-time to begin with.

Reply:
Very good question! C++14 relaxes what can be done inside constexpr functions. With C++14 you can do if statements and for loops (don't remember while loops off the top of my head). You are also allowed to write long functions, whereas in C++11 you were only allowed a single return statement with ternaries. I'm sure you can imagine how gross some functions looked back then :) Because of the increases in constexpr, some people argue every function should be marked constexpr "just in case". With C++20 `new` and `delete` will work at compile time, and so will `std::vector` and `std::string`, which means a tremendous amount of code can be constexpr then. Of course, whether it actually gets evaluated at compile time is a different story. But it will remove more and more of the template metaprogramming in favor of regular programming where the code runs at compile time. The C++ committee is hoping to completely remove the need for template metaprogramming (not templates themselves, though) by around C++26 or C++29. Does that explain things?

Reply:
Excellent, that nicely clarifies for me, thanks.